### PR TITLE
Add option to disable language server

### LINKED ISSUE #4587: Option to disable all IntelliSense services
Don't start the analyzer when option is disabled. A lot of editor features will be unavailable, but giving the user the choice can be useful, if they have a particular need for lower resource usage.

Note that C++ has several IntelliSense related options, including "Disable all IntelliSense features".

https://github.com/Microsoft/PTVS/issues/4582 is a crash where a user attempted to rename the analyzer executable to prevent it from running. That didn't work, it just crashed VS when we tried to launch it.

### PR DESCRIPTION
**Pull request by huguesv**
Fix #4587 

Most code could already handle not getting analysis entry or getting null from an analyzer request, so this is mostly just adding the option, and not starting analysis.

Not entirely satisfied with the option name.

I'm convinced we need this as an escape hatch for problematic projects (I've seen users kill the analyzer process and try to rename it to prevent it from starting), and having the choice between feature and resource usage is good to have.

I'll leave it open until others are back from vacation so we can discuss.

### CODE REVIEW

**Comment by zooba** [comment]
I seem to recall another language having a similar option, that was basically just "Disable IntelliSense".

Having the option sounds good.

**Comment by huguesv** [comment]
We can name the checkbox "Disable all IntelliSense features" like in the C++ option description. I was a bit worried it may not be obvious that this also disables things like refactoring, go to definition, etc. but it's probably ok.

image

File: Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs [hunk 1]
```diff
@@ -241,7 +241,9 @@ internal sealed class IntellisenseController : IIntellisenseController, IOleComm
                 }
 
                 if (entry == null) {
-                    Debug.Fail($"Failed to analyze {buffer.DocumentUri}");
+                    if (!_services.Python.LanguageServerOptions.ServerDisabled) {
```
Comment (zooba) [inline Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs]: Should we just avoid attaching the IntellisenseController completely when disabled?

**Comment by zooba** [comment]
Do we need to add checks to some commands to disable refactoring, etc? Perhaps suppressing the IntellisenseController (and EditFilter?) would do that automatically?

**Comment by MinKPark** [comment]
@huguesv is this PR still needed? if not, how about discarding it?

**Comment by huguesv** [comment]
It's desirable, in the weeks at least 2 people asked for it.

We'll need to see if we can maintain that option when we use Language Server Client (probably).

We can probably bypass the IntelliSense controller as suggested. We'll do another PR if/when we schedule this work.

**Pull request closed without merging**